nixos/modules/installer/cd-dvd/iso-image.nix

```diff
@@ -425,7 +425,8 @@ in
     };
 
     isoImage.squashfsCompression = mkOption {
-      default = "xz -Xdict-size 100%";
+      # FIXME make Xbcj targetSystem dependent once we ship ISOs that contain non-x86 code
+      default = "xz -Xdict-size 100% -Xbcj x86";
```

Comment:
Should we add an assertion that this is x86 to make the failure more obvious?

Reply:
I just realised that it would be trivial to make this optional anyways, I'll do that later.